### PR TITLE
refactor(app): Migrate error recovery utils

### PR DESCRIPTION
**Pull request by mjhuff**
# Overview

This PR splits up error recovery utils into separate files in a util directory. This was a TODO and will also help aid in code clarity when we add even more utils during the drop tip wizard refactor. 

I also missed a couple cleanup comments in #15277, so I added those minor changes here. 

# Test Plan
- Verify CI passes. There are no functional changes.

# Risk assessment
low

### CODE REVIEW

File: app/src/organisms/ErrorRecoveryFlows/ErrorRecoveryWizard.tsx [hunk 1]
```diff
@@ -17,12 +18,9 @@ import type { FailedCommand, IRecoveryMap, RecoveryContentProps } from './types'
 import type {
   useRouteUpdateActions,
```
Comment (koji) [inline app/src/organisms/ErrorRecoveryFlows/ErrorRecoveryWizard.tsx:19]: probably 

import { useRouteUpdateActions } from './utils'

Comment (mjhuff) [reply app/src/organisms/ErrorRecoveryFlows/ErrorRecoveryWizard.tsx:19]: It's just used as a type here 👍 

The linter will actually complain if you try to import the fn.
Screenshot 2024-05-29 at 12 38 38 PM

File: app/src/organisms/ErrorRecoveryFlows/utils/useErrorName.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import type { ErrorKind } from '../types'
+
+// Returns the user-facing name of the errorKind.
+export function useErrorName(errorKind: ErrorKind): string {
+  const { t } = useTranslation('error_recovery')
+
+  switch (errorKind) {
+    default:
+      return t('general_error')
+  }
```
Comment (koji) [inline app/src/organisms/ErrorRecoveryFlows/utils/useErrorName.ts:12]: question 
using `switch` because a few error kinds will be added?

Comment (mjhuff) [reply app/src/organisms/ErrorRecoveryFlows/utils/useErrorName.ts:12]: Yep!

**Review by koji — APPROVED** [review]
LGTM!

**Pull request merged**